src/features/tag/hooks/useTagManager.tsx

```diff
@@ -0,0 +1,32 @@
+import { useState, useMemo } from "react";
+
+type UseTagManagerProps = {
+  initialTags: string[];
+  availableTags: string[];
+};
+
+function useTagManager({ initialTags, availableTags }: UseTagManagerProps) {
```

Comment:
The naming here is a bit confiuing. From reding the function name, i am not sure what is going on below. Do we wneed to extract this in the first place?

Reply:
Agreed. Naming is not ideal. In essence this customhook manages, which tags we show as selectable in the popover, it handles the state of which tags the user has selected and the creation of new tags.

Should we reuse parts of the logic of the tags for other components having UI logic in TagManager and business logic is useTagManager separated might be beneficial. Also should we decide to implement testing for this feature, separating parts of the logic in a custom hook will make testing easier.
Currently as we are not reusing this I agree it adds only overhead and makes it harder to understand what actually happens in the component.
What do you think?